src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -3006,5 +3089,40 @@ public static string ToHexString(ReadOnlySpan<byte> bytes)
 
             return HexConverter.ToString(bytes, HexConverter.Casing.Upper);
         }
+
+
+        /// <summary>
+        /// Converts a span of 8-bit unsigned integers to its equivalent span representation that is encoded with uppercase hex characters.
+        /// </summary>
+        /// <param name="source">A span of 8-bit unsigned integers.</param>
+        /// <param name="destination">The span representation in hex of the elements in <paramref name="source"/>.</param>
+        /// <param name="charsWritten">When this method returns, contains the number of chars that were written in <paramref name="destination"/>.</param>
+        /// <returns>true if the conversion was successful; otherwise, false.</returns>
+        public static bool TryToHexString(ReadOnlySpan<byte> source, Span<char> destination, out int charsWritten)
+        {
+            int length = source.Length;
+
+            if (length > int.MaxValue / 2)
+                goto FalseResult;
+
+            if (length == 0)
+            {
+                charsWritten = 0;
+                return true;
+            }
+
+            int requiredCharCount = length * 2;
+
+            if (destination.Length < requiredCharCount)
+                goto FalseResult;
+
+            HexConverter.EncodeToUtf16(source, destination);
+            charsWritten = requiredCharCount;
+            return true;
+
+        FalseResult:
+            charsWritten = 0;
+            return false;
+        }
```

Comment:
We should not be using `goto` unless it's backed by a data that it helps with important, common, real-world scenario performance.

Such micro-optimizations should be performed on a compiler level.

```suggestion
        {
            if (source.Length == 0)
            {
                charsWritten = 0;
                return true;
            }
            else if (source.Length > int.MaxValue / 2 || destination.Length > source.Length  * 2)
            {
                charsWritten = 0;
                return false;
            }

            HexConverter.EncodeToUtf16(source, destination);
            charsWritten = source.Length  * 2;
            return true;
        }
```